application_sdk/clients/redis.py

```diff
@@ -0,0 +1,236 @@
+"""Redis client for distributed locking with high availability support."""
+
+from typing import Optional
+
+import redis
+from redis.sentinel import Sentinel
+
+from application_sdk.constants import (
+    REDIS_CONNECTION_POOL_SIZE,
+    REDIS_DB,
+    REDIS_HOST,
+    REDIS_PASSWORD,
+    REDIS_PORT,
+    REDIS_SENTINEL_HOSTS,
+    REDIS_SENTINEL_SERVICE_NAME,
+    REDIS_SOCKET_TIMEOUT,
+    STRICT_LOCKING_ENABLED,
+)
+from application_sdk.observability.logger_adaptor import get_logger
+
+logger = get_logger(__name__)
+
+
+class DistributedLock:
+    """Context manager for distributed locks with automatic cleanup."""
+
+    def __init__(
+        self, client: "RedisClient", resource_id: str, owner_id: str, ttl: int
+    ):
+        self.client = client
+        self.resource_id = resource_id
+        self.owner_id = owner_id
+        self.ttl = ttl
+        self.acquired = False
+
+    def __enter__(self) -> bool:
+        """Attempt to acquire the distributed lock."""
+        self.acquired = self.client._acquire_lock(
+            self.resource_id, self.owner_id, self.ttl
+        )
+        return self.acquired
+
+    def __exit__(self, exc_type, exc_val, exc_tb):
+        """Release the lock if it was acquired."""
+        if self.acquired:
+            self.client._release_lock(self.resource_id, self.owner_id)
+
+
+class RedisClient:
```

Comment:
The `RedisClient` class has both the client functionality and the lock functionality bundled together, let's separate them out.